assets/js/controllers/buy.controller.js

```diff
@@ -53,17 +53,19 @@ function BuyCtrl ($scope, $filter, $q, MyWallet, Wallet, MyWalletHelpers, Alerts
   $scope.goTo('amount');
 
   $scope.formattedTrade = undefined;
-  $scope.bitcoinReceived = bitcoinReceived || $scope.trade && $scope.trade.bitcoinReceived;
+  $scope.bitcoinReceived = buyOptions.bitcoinReceived && $scope.trade && $scope.trade.bitcoinReceived;
 
   $scope.fields = { email: $scope.user.email, countryCode: $scope.exchange.profile.country };
-  let txTemplate = { fiat: 0, btc: 0, fee: 0, total: 0, currency: buySell.getCurrency($scope.trade) };
-  $scope.transaction = Object.assign({}, txTemplate, transaction);
+
+  $scope.transaction = trade == null
+    ? ({ fiat: buyOptions.fiat || 0, btc: 0, fee: 0, total: 0, currency: buyOptions.currency || buySell.getCurrency() })
+    : ({ fiat: $scope.trade.inAmount / 100, btc: 0, fee: 0, total: 0, currency: buySell.getCurrency($scope.trade) });
```

Comment:
@plondon / @jtormey I suggest adding or changing a filter to convert cents to dollars in the UI. That way you can pass around integer values, making rounding issues less likely.